tools/templates/event.go.tmpl

```diff
@@ -41,7 +41,7 @@ var (
 
 type {{.Subject}}{{.Predicate}}SubjectContent  struct{
 {{ range $i, $field := .Contents }}
-	{{ .Name }} {{ .Type }} `json:"{{ .NameLower }}{{ if not .Required }},omitempty{{ end }}"{{ if eq .Name "ArtifactId" }} validate:"purl"{{ end }}`
+	{{ .Name }} {{ .Type }} `json:"{{ .NameLower }}{{ if not .Required }},omitempty{{ end }}"{{ if eq .Name "ArtifactId" }} validate:"purl"{{ end }} cdevents:"{{ .NameLower }}"`
```

Comment:
yes, we tried using `json` tag info for copygen, but the same `json` tag will also be used for serialize/deserialize and this will create a problem in matching the field names for different sources,

For Example the source ProjectCreated Gerrit event,

```
type ProjectCreated struct {
	ProjectName string `json:"projectName" cdevents:"name"`
	HeadName    string `json:"headName" cdevents:"subject_id"`
	Type           string  `json:"type"`
	EventCreatedOn float64 `json:"eventCreatedOn"`
	RepoURL        string  `json:"repoURL,omitempty" cdevents:"context_source"`
}
```

here the `json` tag names projectName, headName are used to serialize gerrit event.

Reply:
I think I'm missing some context or maybe knowledge about copygen to better understand what's going on.

I understand that `json` tag in types that represent Gerrit events is used to marshal/unmarshal gerrit events to JSON - that's the standard way of doing JSON marshalling/unmarshalling in go.

I guess `cdevents` tag is meant to tell `copygen` that it should fill the field of a struct with the CDEvents field specified in the tag. What I don't understand is why we need to add `cdevents` tags on the CDEvents SDK side.

Reply:
Actually the same `tag` should be present in all the structs(Gerrit + CDEvents) to match fields and generate functions to copy fields from Gerrit event to CDEvent. 
So we need to have a tag names on CDEvents SDK and the same tag names can be used to match different source events, which can be translated into CDEvents.

An example from copygen shows model struct copied into domain struct, https://github.com/switchupcb/copygen/tree/main/examples/tag#example-tag